src/core/thread/mle.hpp

```diff
@@ -1980,15 +1960,13 @@ class Mle : public InstanceLocator, private NonCopyable
     bool        IsDetachingGracefully(void) { return mDetachGracefullyTimer.IsRunning(); }
     Error       SendChildUpdateRequest(bool aAppendChallenge, uint32_t aTimeout);
 
-#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE || OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
-    Error SendDataRequest(const Ip6::Address                        &aDestination,
-                          const uint8_t                             *aTlvs,
-                          uint8_t                                    aTlvsLength,
-                          uint16_t                                   aDelay,
-                          const LinkMetrics::LinkMetrics::QueryInfo *aQueryInfo = nullptr);
-#else
     Error SendDataRequest(const Ip6::Address &aDestination, const uint8_t *aTlvs, uint8_t aTlvsLength, uint16_t aDelay);
-#endif
+    Error SendDataRequest(const Ip6::Address &aDestination,
+                          const uint8_t      *aTlvs,
+                          uint8_t             aTlvsLength,
+                          uint16_t            aDelay,
+                          const uint8_t      *aExtraTlvsBuf,
+                          uint8_t             aExtraTlvsBufLength);
```

Comment:
I am not fond of the changes here to `SendDataRequest()` method.
- We have `aTlvs/aTlvsLength` and then `aExtraTlvsBuf` and `aExtraTlvsBufLength` (reader of code cannot tell what they are and how they are different) 
- If we use `LinkMetrics::LinkMetrics::QueryInfo *aQueryInfo` it is clear that this param is related to `LinkMetrcis` vs `aExtraTlvsBuf`?
- When using this method, in `LinkMetrics` we need to prepare the extra TLVs in a local (stack allocated) buffer, instead of having `LinkMetrics` module append the info to the message directly.